lib/membrane_rtmp_plugin/rtmp/source/source.ex

```diff
@@ -1,148 +1,161 @@
 defmodule Membrane.RTMP.Source do
   @moduledoc """
-  Membrane Element for receiving RTMP streams. Acts as a RTMP Server.
-  This implementation is limited to only AAC and H264 streams.
+  Membrane Element for receiving an RTMP stream. Acts as a RTMP Server.
 
-  Implementation based on FFmpeg
+  Upon initialization, the source sends `:rtmp_source_initialized` notification, upon which it should be granted the control over the `socket` via `:gen_tcp.controlling_process/2`.
+  This implementation is limited to only AAC and H264 streams.
   """
   use Membrane.Source
 
   require Membrane.Logger
 
-  alias __MODULE__.Native
-  alias Membrane.{AVC, Buffer, Time}
-
-  def_output_pad :audio,
-    availability: :always,
-    caps: Membrane.AAC.RemoteStream,
-    mode: :pull
+  alias Membrane.RTMP.{Handshake, MessageHandler, MessageParser}
 
-  def_output_pad :video,
+  def_output_pad :output,
     availability: :always,
-    caps: Membrane.H264.RemoteStream,
+    caps: Membrane.RemoteStream,
     mode: :pull
 
-  def_options url: [
-                spec: binary(),
+  def_options socket: [
+                spec: :gen_tcp.socket(),
                 description: """
-                URL on which the FFmpeg instance will be created
+                Socket on which the source will be receiving the RTMP stream. The socket must be already connected to the RTMP client and be in non-active mode (`active` set to `false`).
                 """
               ],
-              timeout: [
-                spec: Time.t() | :infinity,
-                default: :infinity,
+              validator: [
+                spec: Membrane.RTMP.StreamValidator,
                 description: """
-                Time the server will wait for a connection from the client
-
-                Duration given must be a multiply of one second or atom `:infinity`.
+                A Module implementing `Membrane.RTMP.MessageValidator` behaviour, used for validating the stream.
                 """
               ]
 
+  @typedoc """
+  Notification sent when the RTMP Source element is initialized and it should be granted control over the socket using `:gen_tcp.controlling_process/2`.
+  """
+  @type rtmp_source_initialized_t() :: {:rtmp_source_initialized, :gen_tcp.socket(), pid()}
```

Comment:
I think that we are missing other notifications here. I would also mention those types in the `moduledoc` as this is the first thing that people will read and they may skip some types that are not linked anywhere.

Reply:
Also I'm not sure if we need the `rtmp_` prefixes in notifications. Any other bin/pipeline will know that source of the notification.

Another thing, how about this particular notification being named `socket_control_needed`?